derive/params.go

```diff
@@ -0,0 +1,58 @@
+//  Copyright 2021 Jake Son
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package derive
+
+import (
+	"go/types"
+	"strconv"
+	"strings"
+)
+
+var blackIdentifier = "_"
+
+// RenameBlankIdentifier returns a signature that all black identified are renamed.
+func RenameBlankIdentifier(sig *types.Signature, prefixs ...string) *types.Signature {
+	prefix := "param_"
+	if len(prefixs) > 0 {
```

Comment:
Maybe it is ok to have two functions
```
RenameBlankIdentifier(sig *types.Signature)
```
and
```
RenameBlankIdentifierWith(sig *types.Signature, prefix string)
```

Simply because we do not support a list of prefixes and that why the type system makes it very clear
What do you think?

Reply:
Yes, you are right. I think It would be better to make type system clear.